config/versions/versions.go

```diff
@@ -0,0 +1,202 @@
+/*
+versions handles config upgrades and downgrades
+
+  - Versions must be stateful, and not rely upon type definitions in the config pkg. Instead versions must localise types to avoid issues with subsequent changes
+
+  - Versions must upgrade to the next version. Do not retrospectively change versions to match new type changes. Create a new version
+
+  - Versions must be registered in import.go
+*/
+package versions
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/buger/jsonparser"
+	"github.com/thrasher-corp/gocryptotrader/common"
+)
+
+var (
+	errRegisteringVersion  = errors.New("error registering config version")
+	errVersionIncompatible = errors.New("version does not implement ConfigVersion or ExchangeVersion")
+	errVersionSequence     = errors.New("version registered out of sequence")
+	errModifiyingExchange  = errors.New("error modifying exchange config")
+	errNoVersions          = errors.New("error retrieving latest config version: No config versions are registered")
+	errApplyingVersion     = errors.New("error applying version")
+)
+
+// DisabledVersion allows authors to rollback changes easily during development
+type DisabledVersion interface {
+	Disabled() bool
+}
```

Comment:
This needs handling or RM until needed

Reply:
Oooh. Thanks for that catch. I lost some code here.

Reply:
Fixed a5bb916c7 🙏